### PR TITLE
feat: implement console ext

### PR DESCRIPTION
**Pull request by LordOfPolls**
## About

This PR adds a new extension called "console" to the builtin collection. The Console extension allows users to start the bot with the aiomonitor console active, which provides a REPL for the bot.

To access the console, users can connect to the port specified in the constructor (by default, port 501). On Linux, users can do this with nc localhost 501, while on Windows they can use telnet localhost 501. Alternatively, users can also use python -m aiomonitor.cli -p 501 on both platforms.

## Checklist

- [x] The ``pre-commit`` code linter has been run over all edited files to ensure the code is linted.
- [x] I've ensured the change(s) work on `3.8.6` and higher.
- [ ] I have added the `versionadded`, `versionchanged` and `deprecated` to any new or changed user-facing function I committed.

### Pull-Request specification

I've made this pull request: (check all that apply)
  - [ ] For the documentation
  - [x] To add a new feature
  - [ ] As a general enhancement
  - [ ] As a refactor of the library/the library's code
  - [ ] To fix an existing bug
  - [ ] To resolve #ISSUENUMBER

This is:
  - [ ] A breaking change

### CODE REVIEW

File: interactions/ext/console.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    For both platforms you can also use "python -m aiomonitor.cli -p 501" as a replacement for the above commands.
+
+    Args:
+        port: The port to start the aiomonitor on
+        console_port: The port to start the aiomonitor console on
+        **kwargs: The locals to make available in the console, by default this includes `client`, `bot` and `interactions`
+
+    """
+
+    def __init__(self, client: interactions.Client, port: int = 501, console_port: int = 502, **kwargs) -> None:
```
Comment (silasary) [inline interactions/ext/console.py:27]: Can you add a `setup` function that exposes these parameters?

Comment (LordOfPolls) [reply interactions/ext/console.py:27]: Why? The setup hook isn't required to pass arguments to an extension?

`bot.load_extension("interactions.ext.console", port=4000)` works fine without a setup hook

Comment (silasary) [reply interactions/ext/console.py:27]: Oh, I must have missed that. Neat

**Review by silasary — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -18,10 +18,12 @@ emoji = "^2.1.0"
 # Optional dependencies
 orjson  = {version = "^3.6.8", optional = true}
 jurigged = {version = "^0.5.3", optional = true}
+aioconsole = {version= "^0.6.0", optional = true}
```
Comment (i0bs) [inline pyproject.toml]:
```suggestion
aioconsole = {version = "^0.6.0", optional = true}
```
bad juju

**Review by i0bs — APPROVED** [review]

**Pull request merged**